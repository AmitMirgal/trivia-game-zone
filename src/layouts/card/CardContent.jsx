import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const Content = styled.div`
  display: flex;
  flex-direction: ${(props) => (props.row ? "row" : "column")};
  font-size: ${(props) => (props.fontSize ? props.fontSize : "24px")};
  line-height: ${(props) => (props.lineHeight ? props.lineHeight : "30px")};
  padding: ${(props) => (props.padding ? props.padding : "12px")};
  margin: ${(props) => (props.padding ? props.padding : "12px")};
`;

const CardContent = (props) => {
  const { children, fontSize, lineHeight, rowAlign } = props;

  return (
    <Content fontSize={fontSize} lineHeight={lineHeight} rowAlign={rowAlign}>
      {children}
    </Content>
  );
};

CardContent.propTypes = {
  children: PropTypes.node,
  fontSize: PropTypes.string,
  lineHeight: PropTypes.string,
  rowAlign: PropTypes.bool,
};

export default CardContent;
