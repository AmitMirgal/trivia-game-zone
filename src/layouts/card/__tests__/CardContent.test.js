import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import CardContent from "../CardContent";

describe("<CardContent/>", () => {
  let wrapper;
  const prepare = () => {
    wrapper = shallow(<CardContent />);
  };

  it("renders CardContent component", () => {
    // when
    prepare();

    // then
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("count number of props", () => {
    // when
    prepare();

    const propsSize = Object.keys(wrapper.props()).length;

    // then
    expect(propsSize).toBe(4);
  });
});
