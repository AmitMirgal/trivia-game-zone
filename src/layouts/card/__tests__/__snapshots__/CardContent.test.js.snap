// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardContent/> renders CardContent component 1`] = `<styled.div />`;
